pkg/controllers/jobset_controller.go

```diff
@@ -316,13 +436,24 @@ func (r *JobSetReconciler) shouldCreateJob(jobName string, ownedJobs *childJobs)
 	return true
 }
 
-// updateStatus updates the status of a JobSet by appending
-// the new condition to jobset.status.conditions.
-func (r *JobSetReconciler) updateStatus(ctx context.Context, js *jobset.JobSet, condition metav1.Condition) error {
+// updateStatus updates the status of a JobSet.
+func (r *JobSetReconciler) updateStatus(ctx context.Context, js *jobset.JobSet, eventReason, eventMsg string) error {
```

Comment:
add eventType string
```suggestion
func (r *JobSetReconciler) updateStatus(ctx context.Context, js *jobset.JobSet, eventType, eventReason, eventMsg string) error {
```